### PR TITLE
Add configuration catalogue for cc

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
## Purpose
Fix https://github.com/wso2-enterprise/choreo/issues/4163

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: en/tools/config-catalog-generator-cc/data/adapter/configs.json [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+      ],
+      "exampleFile": "keystore.toml"
+    },
+    {
+      "title": "Truststore",
+      "options": [
+        {
+          "name": "truststore",
+          "required": true,
+          "description": "The configurations of trust store used in in the Choreo Connect Adapter.",
```
Comment (ashera96) [inline en/tools/config-catalog-generator-cc/data/adapter/configs.json:84]: Typo in description -> in in
Also, the use of full stop at the end of descriptions is not consistent across the configs.json files. Isn't it better to maintain some sort of consistency?

File: en/tools/config-catalog-generator-cc/data/adapter/configs.json [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+              "required": true,
+              "default": "false",
+              "possible": "true, false",
+              "description": "Enable/Disable consul service."
+            },
+            {
+              "name": "url",
+              "type": "string",
+              "required": true,
+              "description": "connection url of the Consul HTTP API."
```
Comment (ashera96) [inline en/tools/config-catalog-generator-cc/data/adapter/configs.json:143]: Start of description with simple letter -> `Connection url of ...`

File: en/tools/config-catalog-generator-cc/data/router/configs.json [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+          "name": "router",
+          "required": true,
+          "description": "The configurations required for router to route the traffic from different clients to services.",
+          "params": [
+            {
+              "name": "listenerHost",
+              "type": "string",
+              "required": true,
+              "default": "0.0.0.0",
+              "description": "host for Listener of Router."
```
Comment (ashera96) [inline en/tools/config-catalog-generator-cc/data/router/configs.json:16]: `host for Listener ... ` -> `Host for listener ... `

File: en/tools/config-catalog-generator-cc/data/router/configs.json [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+      ],
+      "exampleFile": "router.toml"
+    },
+    {
+      "title": "Keystore",
+      "options": [
+        {
+          "name": "keystore",
+          "required": true,
+          "description": "The configurations of key store used in Choreo Connect Adapter",
```
Comment (ashera96) [inline en/tools/config-catalog-generator-cc/data/router/configs.json:71]: This description is the same as the description used for [adapter.keystore]. Shouldn't it be altered as follows: `The configurations of key store used in Choreo Connect Router`?

File: en/tools/config-catalog-generator-cc/data/router/configs.json [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+          "params": [
+            {
+              "name": "certPath",
+              "type": "string",
+              "required": true,
+              "default": "/home/wso2/security/keystore/mg.pem",
+              "description": "Path of the certificate of the Adaptor"
```
Comment (ashera96) [inline en/tools/config-catalog-generator-cc/data/router/configs.json:78]: `of the Adapter` or `of the Router`?

File: en/tools/config-catalog-generator-cc/data/router/configs.json [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+            },
+            {
+              "name": "keyPath",
+              "type": "string",
+              "required": true,
+              "default": "/home/wso2/security/keystore/mg.key",
+              "description": "Path of the private key of the Adaptor"
```
Comment (ashera96) [inline en/tools/config-catalog-generator-cc/data/router/configs.json:85]: `of the Adapter` or `of the Router`?

File: en/tools/config-catalog-generator-cc/data/router/configs.json [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+              "type": "boolean",
+              "required": true,
+              "possible": "true, false",
+              "description": "Enable CORS configurations globally for all endpoints and APIs deployed in Choreo Connect Router"
+            },
+            {
+              "name": "allowOrigins",
+              "type": "list of strings",
+              "required": true,
+              "description": "Allowed origins. set this to [*] allow all origins."
```
Comment (ashera96) [inline en/tools/config-catalog-generator-cc/data/router/configs.json:111]: Suggestion:
`Allowed origins. set this to [*] allow all origins.` -> `Allowed origins. Set this to [*] to allow all origins.`

File: en/tools/config-catalog-generator-cc/data/enforcer/configs.json [hunk 8]
```diff
@@ -0,0 +1,721 @@
...
+              "required": true,
+              "default": "10000",
+              "description": "Maximum cache size."
+            },
+            {
+              "name": "expiryTime",
+              "type": "integer",
+              "required": true,
+              "default": "15",
+              "description": "Expiry time in seconds"
```
Comment (ashera96) [inline en/tools/config-catalog-generator-cc/data/enforcer/configs.json:146]: `in seconds` or `in minutes`?

File: en/tools/config-catalog-generator-cc/data/enforcer/configs.json [hunk 9]
```diff
@@ -0,0 +1,721 @@
...
+              "required": true,
+              "default": "false",
+              "description": "Remove authorization header from backend request"
+            },
+            {
+              "name": "authorizationHeader",
+              "type": "string",
+              "required": true,
+              "default": "authorization",
+              "description": "Header name which authorization token coming from the downstream client"
```
Comment (ashera96) [inline en/tools/config-catalog-generator-cc/data/enforcer/configs.json:381]: Is this description grammatically correct? `... which authorization token ... `?

File: en/tools/config-catalog-generator-cc/data/adapter/configs.json [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+          "required": true,
+          "description": "This includes configurations required for configuring the deployment parameters that are used for identifying the Choreo Connect Adapter REST APIs.",
+          "params": [
+            {
+              "name": "enabled",
+              "type": "boolean",
+              "required": true,
+              "default": "true",
+              "possible": "true, false",
+              "description": "Enable/Disable Choreo Connect Adaptor REST API."
```
Comment (ashera96) [inline en/tools/config-catalog-generator-cc/data/adapter/configs.json:17]: Typo across many descriptions with the usage of the word `Adaptor`. Replace all instances with `Adapter`.

File: en/tools/config-catalog-generator-cc/data/enforcer/configs.json (hunk 8 above)
Comment (AmaliMatharaarachchi) [reply en/tools/config-catalog-generator-cc/data/enforcer/configs.json:146]: minutes. fixed

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**